GoodMorning/GoodMorning/API/OpenWeatherAPI.swift

```diff
@@ -0,0 +1,46 @@
+//
+//  OpenWeatherAPI.swift
+//  GoodMorning
+//
+//  Created by Miro on 2023/07/20.
+//
+
+import Foundation
+
+// 아래는 apiKey가 정의되었다고 가정
+extension Bundle {
+    static var openWeatherAPIKey = "key"
+}
+
+enum OpenWeatherAPI {
+
+    case currentWeather(Coordinate)
+
+    static var baseURL: String {
+        return "https://api.openweathermap.org"
+    }
+
+    var path: String {
+        switch self {
+        case .currentWeather:
+            return "/data/2.5/weather"
+        }
+    }
+
+    var query: [URLQueryItem] {
+        var queryItem = [URLQueryItem]()
+        let keyParameter = URLQueryItem(name: "appid", value: Bundle.openWeatherAPIKey)
+        queryItem.append(keyParameter)
+
+        switch self {
+        case .currentWeather(let coordinate):
+            let latitudeParameter = URLQueryItem(name: "lat", value: coordinate.latitude)
+            let longitudeParameter = URLQueryItem(name: "lon", value: coordinate.longitude)
+            let unitParameter = URLQueryItem(name: "units", value: "metric")
+            queryItem.append(contentsOf: [latitudeParameter, longitudeParameter, unitParameter])
+        }
+
+        return queryItem
```

Comment:
현재 프로젝트에는 Coordinate 라는 타입을 생성했어요. Coordinate type 자체에 URLQueryItem 속성을 추가하는 것도 괜찮겠다는 생각이 들었어요. Coordinate는 모델 타입이고, 해당 모델 타입을 사용한 URLQueryItem이니 계산속성으로 정의하는 방법은 어떨까요?

```swift
struct Coordinate: Decodable {

    let longitude: String
    let latitude: String

    var urlQueryItem: [URLQueryItem] {
        return [
            URLQueryItem(name: "lat", value: latitude),
            URLQueryItem(name: "lon", value: longitude)
        ]
    }
}
```
```suggestion
var query: [URLQueryItem] {
        var queryItem = [URLQueryItem]()
        let keyParameter = URLQueryItem(name: "appid", value: Bundle.openWeatherAPIKey)
        queryItem.append(keyParameter)

        switch self {
        case .currentWeather(let coordinate):
            queryItem.append(contentsOf: coordinate.urlQueryItem)
            let unitParameter = URLQueryItem(name: "units", value: "metric")
            queryItem.append(unitParameter)
        }

        return queryItem
    }
```

Reply:
앗 저도 처음에 계산속성으로 뺄 생각을 했었는데, Coordinate 타입 자체가 url관련된 값을 가지고 있는 것 보다는 다른 곳에서 Coordinate 값을 활용해서 urlQueryItem을 만드는 게 좋다고 생각했거든요!

근데, 지금 다시보니까 결과적으로 Coordinate 타입은 네트워크를 할 때 활용되는 모델이니까 URL 관련 로직을 가지고 있어도 좋을 거 같네요!

말씀해주신대로 코드 수정하겠습니다~🙇‍♂️